examples/ant/train_hiro_ant.py

```diff
@@ -0,0 +1,258 @@
+    )
+    parser.add_argument(
+        "--eval-interval",
+        type=int,
+        default=10 ** 5,
+        help="Frequency (in timesteps) of evaluation phase.",
+    )
+    parser.add_argument(
+        "--update-interval",
+        type=int,
+        default=1,
+        help="Frequency (in timesteps) of network updates.",
+    )
+    parser.add_argument(
+        "--eval-n-runs",
+        type=int,
+        default=100,
+        help="Number of episodes used for evaluation.",
+    )
+    parser.add_argument(
+        "--log-level",
+        type=int,
+        default=20,
+        help="Logging level. 10:DEBUG, 20:INFO etc.",
+    )
+    parser.add_argument(
+        "--env",
+        default="AntMaze",
+        type=str)
+    parser.add_argument(
+        "--render",
+        action="store_true",
+        default=False,
+        help="Render env states in a GUI window.",
+    )
+    parser.add_argument("--lr", type=float, default=6.25e-5, help="Learning rate")
+    parser.add_argument(
+        "--num-envs", type=int, default=1, help="Number of envs run in parallel."
+    )
+    parser.add_argument(
+        "--batch-size", type=int, default=32, help="Batch size used for training."
+    )
+    parser.add_argument(
+        "--record",
+        action="store_true",
+        default=False,
+        help="Record videos of evaluation envs. --render should also be specified.",
+    )
+    parser.add_argument("--gamma", type=float, default=0.99, help="Discount factor.")
+    args = parser.parse_args()
+
+    import logging
+
+    logging.basicConfig(level=args.log_level)
+
+    # Set a random seed used in PFRL.
+    utils.set_random_seed(args.seed)
+
+    # Set different random seeds for different subprocesses.
+    # If seed=0 and processes=4, subprocess seeds are [0, 1, 2, 3].
+    # If seed=1 and processes=4, subprocess seeds are [4, 5, 6, 7].
+    process_seeds = np.arange(args.num_envs) + args.seed * args.num_envs
+    assert process_seeds.max() < 2 ** 32
+
+    args.outdir = experiments.prepare_output_dir(args, args.outdir)
+    print("Output files are saved in {}".format(args.outdir))
+
+    def make_ant_env(idx, test):
+
+        # use different seeds for train vs test envs
+        process_seed = int(process_seeds[idx])
+        env_seed = 2 ** 32 - 1 - process_seed if test else process_seed
+        utils.set_random_seed(env_seed)
+        env = AntEnvWithGoal(create_maze_env(args.env), args.env)
+
+        env.seed(int(env_seed))
+
+        if test and args.record:
+            assert args.render, "To use --record, --render needs be specified."
+            video_dir = os.path.join(args.outdir, "video_{}".format(idx))
+            os.mkdir(video_dir)
+            env = RecordMovie(env, video_dir)
+        return env
+
+    def make_batch_ant__env(test):
+        return pfrl.envs.MultiprocessVectorEnv(
+            [functools.partial(make_ant_env, idx, test) for idx in range(args.num_envs)]
+        )
```

Comment:
That's a separate design pattern that we should address with the maintainers of the repo in a separate PR, since that would require more changes to more than this file.